src/app.rs

```diff
@@ -169,6 +171,20 @@ pub enum TimeFrame {
 }
 
 impl TimeFrame {
+    pub fn duration(self) -> Option<Duration> {
```

Comment:
This part suck, do you have any idea?

Reply:
Why does it suck? This is the idiomatic way of doing it in Rust, as far as I can tell. You can't easily attach constant properties to enums... I mean there's https://docs.rs/strum/0.18.0/strum/trait.EnumProperty.html but...

Reply:
I mean the way I calc the duration is inaccurate.

Reply:
I consider 30 days = 1 month, do we have better way of doing?

Reply:
Oh, I think it's correct. Consider that trading days don't even match with calendar months in the first place. For example, no trading on weekends and public holidays.